pygis/docs/e_extraction.md

```diff
@@ -0,0 +1,350 @@
+* [Creating Geospatial Vector Data](e_new_vectors.md)
+```
+----------------
+
+# Extraction
+
+Subsetting and extracting data is useful when we want to select or analyze a portion of the dataset based on a feature's location, attribute, or its spatial relationship to another dataset.
+
+In this chapter, we will explore three ways that data from a GeoDataFrame can be subsetted and extracted: clip, select by attribute, and select by location.
+
+## Setup
+
+First, let's import the necessary modules.
+
+```{code-cell} ipython3
+# Import modules
+import matplotlib.pyplot as plt
+import pandas as pd
+import geopandas as gpd
+from shapely.geometry import Polygon
+```
+
+We will utilize shapefiles of San Francisco Bay Area county boundaries and wells within the Bay Area and the surrounding 50 km. We will first load in the data and reproject the data.
+
+```{important} All the data must have the same coordinate system in order for extraction to work correctly.
+```
+
+```{code-cell} ipython3
+# Load data
+
+# County boundaries
+# Source: https://opendata.mtc.ca.gov/datasets/san-francisco-bay-region-counties-clipped?geometry=-125.590%2C37.123%2C-119.152%2C38.640
+counties = gpd.read_file("../_static/e_vector_shapefiles/sf_bay_counties/sf_bay_counties.shp")
+
+# Well locations
+# Source: https://gis.data.ca.gov/datasets/3a3e681b894644a9a95f9815aeeeb57f_0?geometry=-123.143%2C36.405%2C-119.230%2C37.175
+# Modified by author so that only the well locations within the counties and the surrounding 50 km were kept
+wells = gpd.read_file("../_static/e_vector_shapefiles/sf_bay_wells_50km/sf_bay_wells_50km.shp")
+
+# Reproject data to NAD83(HARN) / California Zone 3
+# https://spatialreference.org/ref/epsg/2768/
+proj = 2768
+counties = counties.to_crs(proj)
+wells = wells.to_crs(proj)
+```
+
+We will also create a rectangle over a part of the Bay Area.
+
+```{code-cell} ipython3
+# Create list of coordindate pairs
+coordinates = [[1790787, 736108], [1929652, 736108], [1929652, 598414], [1790787, 598414]]
+
+# Create a Shapely polygon from the coordinate-tuple list
+poly_shapely = Polygon(coordinates)
+
+# Create a dictionary with needed attributes and required geometry column
+attributes_df = {'Attribute': ['name1'], 'geometry': poly_shapely}
+
+# Convert shapely object to a GeoDataFrame
+poly = gpd.GeoDataFrame(attributes_df, geometry = 'geometry', crs = "EPSG:2768")
+```
+
+We'll define some functions to make displaying and mapping our results a bit easier.
+
+```{code-cell} ipython3
+def display_table(table_name, attribute_table):
+    '''Display the first and last five rows of attribute table.'''
+
+    # Print title
+    print("Attribute Table: {}".format(table_name))
+
+    # Print number of rows and columns
+    print("\nTable shape (rows, columns): {}".format(attribute_table.shape))
+
+    # Display first five rows of attribute table
+    print("\nFirst five rows:")
+    display(attribute_table.head())
```

Comment:
`display` is a cool trick, didn't know that. however let's just show the top and bottom 2 rows so `.head(2)`. Its overwelming with top and bottom 5.